### ISSUE TITLE
New linter — `:redundant-ignore`

### ISSUE THREAD

**Issue by DeLaGuardo** [clj-kondo/clj-kondo#2366]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**Is your feature request related to a problem? Please describe.**
During library upgrade or adjusting kondo config some forms ignored via `:clj-kondo/ignore` metadata might get "fixed" and stop reporting any linter warnings. In that case it would be great to have a way to know which metadata entry can be removed to keep code clean.

**Describe the solution you'd like**
Would be great to have a linter that reports :info level message in case ignored form no longer reports any misperfections.

**Comment by borkdude** [comment]
Implemented in branch redundant-ignore. 

Came across a false positive that still has to be solved:

```

#_{:clj-kondo/ignore [:unused-private-var]}
(defn- -debug [& strs]
  (.println System/err
            (with-out-str
              (apply println strs))))
```
